[Cherry-Pick] UefiCpuPkg: Enable MM_CORE_STANDALONE for IntelMmSaveStateLib, SmmCpuPlatformHookLibNull

apop5 opened the pull request:
## Description

MmSupervisorPkg's MmSupervisorCore.inf  (MM_CORE_STANDALONE) uses Both SmmCpuPlatformHookLib and MmSaveStateLib.

Edk2 limited the scope of the libraries to not support MM_CORE_STANDALONE, and this works okay for the Edk2's StandaloneMmPkg, but it breaks the expectations of the MmSupervisorPkg's Core.


Add MM_CORE_STANDALONE as an allowed Module Type for these libraries. 

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?
- [x] Backport to release branch?

## How This Was Tested
Building on silicon vendor codebase which consumed MmSupervisorCore.

## Integration Instructions
No integration required.

os-d reviewed (commented):
Instead of going in as reverts of edk2 commits should this go in as MU_CHANGEs?

makubacki commented:
> Instead of going in as reverts of edk2 commits should this go in as MU_CHANGEs?

Also, @apop5, are you going to make the upstream change to the allowed Standalone MM types? If so, I'd rather just make that be the MU_CHANGE since it is closer to what upstream will have.

apop5 commented:
> > Instead of going in as reverts of edk2 commits should this go in as MU_CHANGEs?
> 
> Also, @apop5, are you going to make the upstream change to the allowed Standalone MM types? If so, I'd rather just make that be the MU_CHANGE since it is closer to what upstream will have.

This revert is happening because SupvMM is attempting to consume these libraries.
StandaloneMmPkg does not attempt to consume them. 
My hope is to rework the MM Supv to be more in sync with standalonemmcore, and that these changes can be dropped.

Alternatives are:
Make Stmm versions of these libraries (in the SupvMM package)

I don't think these changes should be upstreamed to edk2, because this is happening because SupvMM is attempting to consume them.

makubacki commented:
> I don't think these changes should be upstreamed to edk2, because this is happening because SupvMM is attempting to consume them.

I think that's why it should be supported. These are in `UefiCpuPkg` which should support all MM implementations. I don't see a reason the upstream needs to restrict the linking module types to exclude Standalone MM for these library instances.

apop5 commented:
> > I don't think these changes should be upstreamed to edk2, because this is happening because SupvMM is attempting to consume them.
> 
> 
> 
> I think that's why it should be supported. These are in `UefiCpuPkg` which should support all MM implementations. I don't see a reason the upstream needs to restrict the linking module types to exclude Standalone MM for these library instances.
> 
> 

I'll get a PR up for edk2

apop5 committed:
[Cherry-Pick] UefiCpuPkg: Add StmmCore for IntelMmSaveStateLib,SmmCpuPlatformHookLibNull.

IntelMmSaveStateLib and SmmCpuPlatformHookLibNull were limited to
DXE_SMM_DRIVER MM_STANDALONE, where as previously not supported module
types were listed.

Add MM_CORE_STANDALONE as a supported module type for these libraries.

Signed-off-by: Aaron Pop <aaronpop@microsoft.com>

apop5 commented:
PR to edk2 is up

https://github.com/tianocore/edk2/pull/10914
